examples/orthogonal_subspace_learning/osf_continual_learning.py

```diff
@@ -0,0 +1,694 @@
+
+    valid_labels = ["Dovish", "Hawkish", "Neutral"]
+
+    with torch.no_grad():
+        for batch in dataloader:
+            input_ids = batch["input_ids"].to(model.device)
+            attention_mask = batch["attention_mask"].to(model.device)
+            labels = batch["labels"]
+
+            outputs = model.generate(
+                input_ids=input_ids,
+                attention_mask=attention_mask,
+                max_new_tokens=10,
+                pad_token_id=tokenizer.pad_token_id,
+                do_sample=False,
+            )
+
+            for i in range(len(outputs)):
+                generated_text = tokenizer.decode(outputs[i], skip_special_tokens=True)
+
+                # Extract sentiment label
+                pred = None
+                for label in valid_labels:
+                    if label in generated_text:
+                        pred = label
+                        break
+
+                # Get ground truth
+                label_ids = labels[i][labels[i] != -100]
+                if len(label_ids) > 0:
+                    gt = tokenizer.decode(label_ids, skip_special_tokens=True).strip()
+                    if pred == gt:
+                        correct += 1
+                    total += 1
+
+    accuracy = correct / total if total > 0 else 0.0
+    return accuracy
+
+
+def evaluate_model(model, eval_dataset, data_collator, tokenizer, task_name, task_type):
+    """Evaluate model on a dataset and return loss and accuracy."""
+    # Compute loss
+    trainer = Trainer(
+        model=model,
+        data_collator=data_collator,
+        eval_dataset=eval_dataset,
+    )
+    results = trainer.evaluate()
+    loss = results["eval_loss"]
+
+    # Compute accuracy based on task type
+    if task_type == "scienceqa":
+        accuracy = compute_accuracy_scienceqa(model, eval_dataset, tokenizer, data_collator)
+    elif task_type == "numglue":
+        accuracy = compute_accuracy_numglue(model, eval_dataset, tokenizer, data_collator)
+    elif task_type == "fomc":
+        accuracy = compute_accuracy_fomc(model, eval_dataset, tokenizer, data_collator)
+    else:
+        accuracy = 0.0
+
+    print(f"  {task_name}: Loss = {loss:.4f}, Accuracy = {accuracy * 100:.2f}%")
+    return loss, accuracy
+
+
+def train_with_osf(
+    model_name,
+    num_train,
+    num_eval,
+    output_dir,
+    num_epochs,
+    learning_rate,
+    batch_size,
+    gradient_accumulation_steps,
+    max_length,
+    seed,
+):
+    """Train using OSF with progressive rank allocation."""
+    print("\n" + "=" * 80)
+    print("TRAINING WITH OSF (Orthogonal Subspace Fine-tuning)")
+    print("=" * 80)
+
+    # Load tokenizer and base model
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+    tokenizer.pad_token = tokenizer.eos_token
+
+    base_model = AutoModelForCausalLM.from_pretrained(
+        model_name, torch_dtype=torch.bfloat16, device_map="auto", trust_remote_code=True
```

Comment:
```suggestion
        model_name, torch_dtype=torch.bfloat16, device_map="auto",
```

this isn't a requirement or is it?